spring-cloud-sleuth-zipkin-stream/src/main/java/org/springframework/cloud/sleuth/zipkin/stream/SamplingZipkinSpanIterator.java

```diff
@@ -138,4 +158,45 @@ public void remove() {
 		}
 		return zipkinSpan.build();
 	}
+
+	private static void addLocalComponentAnnotation(Span span, Builder zipkinSpan,
+			Endpoint ep) {
+		String processId = span.getProcessId() != null
+				? span.getProcessId().toLowerCase()
+				: ZipkinMessageListener.UNKNOWN_PROCESS_ID;
+		zipkinSpan.addBinaryAnnotation(
+				BinaryAnnotation.create(Constants.LOCAL_COMPONENT, processId, ep));
+	}
+
+	private static void addServerAddressAnnotation(zipkin.Span.Builder zipkinSpan,
+			Endpoint ep) {
+		BinaryAnnotation component = new BinaryAnnotation.Builder()
+				.type(BinaryAnnotation.Type.STRING)
+				.key(Constants.SERVER_ADDR)
+				.value(ep.serviceName)
+				.endpoint(ep).build();
+		zipkinSpan.addBinaryAnnotation(component);
+	}
+
+	private static boolean spanContainsAnyZipkinConstant(Span span) {
+		for (org.springframework.cloud.sleuth.Log log : span.logs()) {
+			if (ZIPKIN_ANNOTATIONS.contains(log.getEvent())) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	private static boolean spanContainsAnyZipkinClientConstantAndSaIsNotSet(Span span) {
+		boolean containsAnyZipkinClientConstant = false;
+		for (org.springframework.cloud.sleuth.Log log : span.logs()) {
+			if (ZIPKIN_CLIENT_ANNOTATIONS.contains(log.getEvent())) {
+				containsAnyZipkinClientConstant = true;
```

Comment:
I think logically it would be the same to `return !span.tags().containsKey(Constants.SERVER_ADDR)` here instead of break